tests/components/image_processing/test_facebox_face_detect.py

```diff
@@ -0,0 +1,68 @@
+"""The tests for the facebox component."""
+import requests_mock
+
+from homeassistant.const import (CONF_IP_ADDRESS, CONF_PORT)
+from homeassistant.setup import setup_component
+import homeassistant.components.image_processing as ip
+
+from tests.common import (
+    get_test_home_assistant, assert_setup_component)
+
+MOCK_IP = '192.168.0.1'
+MOCK_PORT = '8080'
+
+MOCK_RESPONSE = """
+{"facesCount": 1,
+"success": True,
+"faces":['face_data']}
+"""
+
+VALID_ENTITY_ID = 'image_processing.facebox_demo_camera'
+VALID_CONFIG = {
+    ip.DOMAIN: {
+        'platform': 'facebox_face_detect',
+        CONF_IP_ADDRESS: MOCK_IP,
+        CONF_PORT: MOCK_PORT,
+        ip.CONF_SOURCE: {
+            ip.CONF_ENTITY_ID: 'camera.demo_camera'}
+        },
+    'camera': {
+        'platform': 'demo'
+        }
+    }
+
+
+class TestFaceboxSetup(object):
```

Comment:
You don't need to use a class for the tests. Just define test functions that have names starting with `test_` Pass in `hass` as parameter to the functions to get a hass instance to test with. Setup and teardown is handled by pytest.

Reply:
Ok